### ISSUE TITLE
Badge is taken from cache and not refreshed when changed

### ISSUE THREAD

**Issue by jcubic** [travis-ci/travis-ci#7115]
I have this case, I have badge in github readme and it says failing, because build was failing, I've fixed the build, refreshes the github readme page and the badge was still saying failing untill I've cleared the cache.

Don't know if this is issue with github or from travis because github is caching images.

**Comment by BanzaiMan** [comment]
Do you have a build log URL that shows the problem you are describing here?

**Comment by jcubic** [comment]
I've created test project that was faling and then passing and when I refreshed github page it's was refreshing, maybe I refreshed page too soon.

**Comment by jcubic** [comment]
But the badge on the site says that build is unknown https://travis-ci.org/jcubic/test

**Comment by jcubic** [comment]
it's fixed when I opened developer tools (I have cache disabled if it's on).

**Comment by BanzaiMan** [comment]
Sorry, but I don't see any reproduction of the issue with this repo.

**Comment by jcubic** [comment]
You will not see the problem because it was my browser that was caching the image, you will need to create a repo with .travis.yml that's failing and when it fail you will see build: unknow (next to repo name) even if it should say failing when you refresh the page, and when you commit that make travis build pass you will not see the change on the site.

**Comment by BanzaiMan** [comment]
So, the browser is caching the image, is it not?

**Comment by jcubic** [comment]
Yes, but it's caching probably because of wrong http headers or something.

**Comment by BanzaiMan** [comment]
Can a server force the client to not cache the image with headers? Not all browsers respects such headers.

http://stackoverflow.com/a/126831/136333 advises adding a string as a parameter (which is summarily discarded by the server). This would work on our hosted services, but will not if the image URL is to be statically embedded (in a README, for example).

**Comment by JanBessai** [comment]
I was confused by this as well. Github has an explanation in their [corresponding issue](https://github.com/github/markup/issues/224#issuecomment-37663375): they cache images to avoid mixed-content warnings. Their caching mechanism respects `Cache-Control: no-cache` headers, so it would be great, if you could add them to the Travis batch.
